src/connection/latestDriverVersionResolver.js

```diff
@@ -0,0 +1,105 @@
+var q = require('q');
+var request = require('request');
+var _ = require('lodash');
+
+function LatestDriverVersionResolver(config, instanceConfig, logger) {
+  this.logger = logger;
+  this.latestVersionRegexp = instanceConfig.latestVersionRegexp;
+}
+
+LatestDriverVersionResolver.prototype.getLatestVersion = function (binary) {
+  var that = this;
+  return that._getLatestMajorVersion(binary)
+    .then(function (result) {
+      if (result.latestMajorVersion) {
+        binary.latestVersionUrl = binary.latestVersionUrl.replace(that.latestVersionRegexp, result.latestMajorVersion);
+      }
+      return that._getLatestDriverVersion(binary);
+    });
+};
+
+LatestDriverVersionResolver.prototype._getLatestMajorVersion = function (binary) {
+  var that = this;
+
+  that.logger.info('Check for latest major version of: ' + binary.filename);
+  return q.Promise(function (resolveFn, rejectFn) {
+    if (binary.latestVersionFileUrl) {
```

Comment:
what about adding testability e.g. be able to specify inline json ?. For example check if the param is a string and treat it as url but if object, use it directly.